### PR TITLE
Add experimental options

### PR DESCRIPTION
**Pull request by jyu00**
### Summary

Add some of the experimental options. 

### Details and comments
Fixes #

### CODE REVIEW

File: qiskit_ibm_runtime/options/resilience_options.py [hunk 1]
```diff
@@ -53,21 +60,53 @@ class ResilienceOptions:
...
+    extrapolator: ExtrapolatorType = None
+
+    # Measurement error mitigation
+    measure_noise_mitigation: Optional[bool] = None
+
+    # ZNE
+    zne_mitigation: Optional[bool] = None
+    zne_noise_factors: Sequence[float] = (1, 3, 5)
+    zne_extrapolator: Optional[Union[ZneExtrapolatorType, Sequence[ZneExtrapolatorType]]] = field(
+        default_factory=lambda: ["linear", "multi_exponential"]
```
Comment (chriseclectic) [inline qiskit_ibm_runtime/options/resilience_options.py:100]: Can you avoid default factor if you use a tuple? Either way the default value should be `("exponetial", "linear")` in that order
```suggestion
        default_factory=lambda: ("exponential", "linear")
```

File: qiskit_ibm_runtime/options/resilience_options.py [hunk 2]
```diff
@@ -36,15 +36,22 @@
     "QuarticExtrapolator",
 ]
 
+ZneExtrapolatorType = Literal[
+    "multi_exponential",
+    "single_exponential",
+    "double_exponential",
+    "linear",
+]
```
Comment (chriseclectic) [inline qiskit_ibm_runtime/options/resilience_options.py:44]: Not sure how you want to handle the `polynomial_degree_k` for 1 <= k <= 7, but you could include the first couple
```suggestion
ZneExtrapolatorType = Literal[
    "exponential",
    "double_exponential",
    "linear",
    "polynomial_degree_1",
    "polynomial_degree_2",
    "polynomial_degree_3",
    "polynomial_degree_4",
]
```

**Review by chriseclectic — CHANGES_REQUESTED** [review]
We should the execution options for `samples`,`shots_per_sample`, and `interleave_samples` too

File: qiskit_ibm_runtime/options/resilience_options.py [hunk 3]
```diff
@@ -85,26 +146,107 @@ def validate_resilience_options(resilience_options: dict) -> None:
...
+                            f"{extrap} requires at least {required_factors[extrap]} zne_noise_factors"
+                        )
+
+            # Validation of threshold
+            threshold = resilience_options.get("zne_stderr_threshold")
+            if threshold is not None and threshold <= 0:
+                raise ValueError("Invalid zne_stderr_threshold option value must be > 0")
+
+        if resilience_options.get("pec_mitigation"):
+            if resilience_options.get("pec_mitigation"):
```
Comment (mberna) [inline qiskit_ibm_runtime/options/resilience_options.py:229]: One of these two options should be zne_mitigation?
```suggestion
        if resilience_options.get("pec_mitigation"):
            if resilience_options.get("zne_mitigation"):
```

**Review by mberna — CHANGES_REQUESTED** [review]

File: qiskit_ibm_runtime/options/options.py [hunk 4]
```diff
@@ -119,23 +132,25 @@ def _get_program_inputs(options: dict) -> dict:
...
-        inputs["resilience_settings"] = options.get("resilience", {})
-        inputs["resilience_settings"].update({"level": options.get("resilience_level")})
-        inputs["run_options"] = options.get("execution")
-        inputs["run_options"].update(
+        inputs["resilience_level"] = options.get("resilience_level")
+        inputs["resilience"] = options.get("resilience", {})
+        inputs["twirling"] = options.get("twirling", {})
+
+        inputs["execution"] = options.get("execution")
+        inputs["execution"].update(
```
Comment (mberna) [inline qiskit_ibm_runtime/options/options.py:153]: are we renaming `run_options` to `execution`?

**Review by mberna — CHANGES_REQUESTED** [review]

**Review by mberna — APPROVED** [review]
LGTM!

**Pull request merged**